Content.Shared/Interaction/SharedInteractionSystem.cs

```diff
@@ -115,6 +120,29 @@ private void OnRemoveAttempt(EntityUid uid, UnremoveableComponent item, Containe
             args.Cancel();
         }
 
+        private bool HandleTryPullObject(ICommonSession? session, EntityCoordinates coords, EntityUid uid)
+        {
+            if (!ValidateClientInput(session, coords, uid, out var userEntity))
+            {
+                Logger.InfoS("system.interaction", $"TryPullObject input validation failed");
+                return true;
+            }
+
+            if (userEntity.Value == uid)
```

Comment:
I know your PR just moves this code, but this `if` condition could use a little comment. I think I know what it does but it's not obvious at a glance.